### DISCUSSION TITLE
Deserialization

### DISCUSSION THREAD

**Discussion by mikecasas** [oqtane/oqtane.framework#784] [Q&A] [answered]
Anyone have a good solution to get around the 'Deserialization of reference types without parameterless constructor is not supported.'?

**Comment by hishamco** [comment]
Is this a generic question or is there something related to Oqtane?

**Reply by mikecasas** [reply]
A little of both, I guess. I was working on a new project and ran into it. I was wondering if someone else ran into this problem.

**Reply by hishamco** [reply]
Newtonsoft has one of the coolest serializer that has a lot of features

> Deserialization of reference types without parameterless constructor is not supported.'?

Ya, this is a limitation, I ran into similar issues when I used System.Text.Json in YesSql, I realize then there 're a lot of features need to be supported into the serializer

**Reply by mikecasas** [reply]
Thanks, that's what I read about.

**Reply by hishamco** [reply]
I think we need to close this while it's a generic question and you could have a look in GitHub I remembered there's another open source serializer that support many missing features, nothing but Newtonsoft is a famous one

**Comment by jimspillane** [comment, marked as answer]
_Support deserializing objects using parameterized constructors_ appears to be added in the upcoming release.
https://github.com/dotnet/runtime/issues/41313
